src/main/java/bookstore/services/authservice/AuthService.java

```diff
@@ -4,26 +4,30 @@
 import bookstore.dtos.user.UserResponseDto;
 import bookstore.exceptions.RegistrationException;
 import bookstore.mappers.user.IUserMapper;
-import bookstore.models.User;
 import bookstore.repositories.userrepository.IUserRepository;
 import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @RequiredArgsConstructor
 @Service
 public class AuthService implements IAuthService {
     private final IUserRepository userRepository;
-
+    private final PasswordEncoder passwordEncoder;
     private final IUserMapper userMapper;
 
     @Override
     public UserResponseDto register(UserRegistrationRequestDto userRegistrationRequestDto) {
-        User user = userRepository.getByEmail(userRegistrationRequestDto.getEmail());
+        boolean isUserExist = userRepository.existsByEmail(userRegistrationRequestDto.getEmail());
 
-        if (user != null) {
+        if (isUserExist) {
```

Comment:
```suggestion
        if (userRepository.existsByEmail(userRegistrationRequestDto.getEmail())) {
```
you can simplify it and remove the variable